lib/screens/explore_screens/regions_screens/regions_main/components/region_event_card/interactor/viewmodels/region_event_card_bloc.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:seeds/datasource/remote/model/profile_model.dart';
+import 'package:seeds/domain-shared/shared_use_cases/get_user_profile_use_case.dart';
+import 'package:seeds/screens/explore_screens/regions_screens/regions_main/components/region_event_card/interactor/mappers/region_event_profiles_state_mapper.dart';
+
+part 'region_event_card_event.dart';
+part 'region_event_card_state.dart';
+
+class RegionEventCardBloc extends Bloc<RegionEventCardEvent, RegionEventCardState> {
+  RegionEventCardBloc() : super(RegionEventCardState.initial()) {
+    on<OnLoadRegionEventMembers>(_onLoadRegionEventMembers);
+  }
+
+  Future<void> _onLoadRegionEventMembers(OnLoadRegionEventMembers event, Emitter<RegionEventCardState> emit) async {
+    if (event.eventUsers.isNotEmpty) {
+      final firstThreeCalls = event.eventUsers.take(3).map((i) => GetUserProfileUseCase().run(i)).toList();
```

Comment:
instead of having this logic here. 

We can have a use case that takes a list of accounts you want to load. 

and pass 3 accounts. 

That way you can move the logic into the use case away from the bloc

Reply:
Moved a use case.